projects/train/train/data/waveforms/generator/cbc.py

```diff
@@ -0,0 +1,131 @@
+import math
+from functools import partial
+from typing import Any, Callable, Dict
+
+import torch
+from train.data.waveforms.generator.generator import WaveformGenerator
+
+
+class CBCGenerator(WaveformGenerator):
+    """
+    A torch module for generating CBC waveforms on the fly using
+    ml4gw waveform models.
+
+    Args:
+    """
+
+    def __init__(
+        self,
+        *args,
+        approximant: Callable,
+        f_min: float = 0.0,
+        f_max: float = 0.0,
+        padding: float = 0.0,
+        waveform_arguments: Dict[str, Any] = None,
+        **kwargs
+    ):
+        """
+        A torch module for generating waveforms on the fly.
+
+        Args:
+            waveform:
+                A callable that returns cross and plus polarizations
+                given a set of parameters.
+            duration:
+                The duration of the waveform in seconds
+            sample_rate:
+                Sample rate of the waveform in Hz
+            f_min:
+                The minimum frequency of the waveform in Hz
+            f_max:
+                The maximum frequency of the waveform in Hz
+            padding:
+                The amount of padding to add to the right
+                of the waveform in seconds
+            waveform_arguments:
+                A dictionary of fixed arguments to pass to the waveform model
+        """
+        super().__init__(*args, **kwargs)
+        waveform_arguments = waveform_arguments or {}
+        self.waveform = partial(approximant, **waveform_arguments)
```

Comment:
What keyword arguments are you passing via `waveform_arguments`?

Reply:
For CBC, I think the only one is `f_ref`. The idea here is to fix certain arguments that aren't randomly sampled

Reply:
OK great. Let's add that under the docstring i.e. `A dictionary of fixed arguments to pass to the waveform model. E.g. ``f_ref`` which stays constant for the run.`